src/lib/stores/search.js

```diff
@@ -14,27 +14,31 @@ export const $recentSearches = map({});
 /**
  * Add search term to recent search history
  * @param {SearchedItem} item 
- * 
- * @todo implement logic to allow holding maximum of 5 words by removing older words when new a one gets added
  */
 export function $addToRecentSearchesFn({ word, url }) {
-  // Re-initialise the state with current localStorage value
-  $recentSearches.set({...JSON.parse(localStorage.getItem("jargons.dev:recent_searches"))});
+  // Re-initialise the state with the current localStorage value
+  const storedSearches = JSON.parse(localStorage.getItem("jargons.dev:recent_searches")) || {};
+  $recentSearches.set({ ...storedSearches });
```

Comment:
```suggestion
  // Re-initialise the state with the current localStorage value
  $recentSearches.set({...JSON.parse(localStorage.getItem("jargons.dev:recent_searches"))});
```

I'd keep this same though, since it works exactly the same way - _if it work I don't touch it_ 😉 

but I like how it made the area more readable though, feel free to not commit this suggestion 😃